src/ObjectGraph.es6.js

```diff
@@ -410,20 +424,29 @@ ObjectGraph.prototype.getType = function(id) {
 // Interface method: Does id refer to a function?
 ObjectGraph.prototype.isFunction = function(id) {
   if ( ! id ) return false;
-
-  // TODO: Should we store a lazy map of these to make this faster?
-  for ( var i = 0; i < this.functions.length; i++ ) {
-    if ( id === this.functions[i] ) return true;
-  }
-
-  return false;
+  return id in this.functions;
 };
 
 // Interface method: Get ids of all objects that are functions.
 ObjectGraph.prototype.getFunctions = function() {
-  return this.functions.slice();
+  return Object.keys(this.functions);
+};
+
+// Interface method: Get the name of given function id.
+ObjectGraph.prototype.getFunctionName = function(id) {
+  return this.functions[id];
+};
+
+// Interface method: get the root of object graph.
+ObjectGraph.prototype.getRoot = function() {
+  return this.root;
 };
 
+// Interface method: Get attribute and id belong to given id.
+ObjectGraph.prototype.getPropertiesIds = function(id){
+  return this.data[id]
```

Comment:
I'm a little anxious about exposing the raw data object for clients to muck with. Perhaps `Object.assign({}, this.data[id])`?